Completed project

Johay90 opened the pull request:
Should be complete with the report challenge implemented (pivot table)

BBualdo commented:
Hello @Johay90 ,

Unfortunatelly, your app has issues. I tried to get stacks when there is no stacks and got and exception, same when trying to seed test stacks.

Please try to find and fix it 😁

If you will stuck, I'm here to help

Johay90 commented:
> Hello @Johay90 ,
> 
> Unfortunatelly, your app has issues. I tried to get stacks when there is no stacks and got and exception, same when trying to seed test stacks.
> 
> Please try to find and fix it 😁
> 
> If you will stuck, I'm here to help

Thanks for the review @BBualdo,

I couldn't reproduce the no stack exceptions you mentioned, and this was something I checked before without running into any issues.

I'm suspecting it might be a database name conflict causing the discrepancies. I've added a suffix to my database name to avoid clashes now. It's not an elegant fix, and making sure our database structures match up during initialization would probably be a better solution.

I also added a few fixes around my input test data/seeding. On a side note, while revisiting this, the code I wrote for seeding became really messy 🤦‍♂️ . I think straight-up manual seeding would have been better for this type of project.

Hopefully, this resolves the issue(s). 🙏 

Thanks!

BBualdo commented:
Hi @Johay90 ,

No, it wasn't fault db naming, because I realised it earlier and renamed it just like you did 😅

Nice work, now it works as it should. I found only one problem when you try to view flashcards in stacks when there is no flashcards 😅

![image](https://github.com/TheCSharpAcademy/CodeReviews.Console.Flashcards/assets/118930386/65a2b4a8-8d41-4db9-b0f0-5cd887f86c59)

And also a small bug, that in seed data there are questions repeated multiple times 😁 But that's due to small number of questions, so It's not a bug that much 😅

I'm approving it anyway, because I see you know what you doin' - services, MVC, Separation of Concerns, really good job!

Sorry for making you waiting 😅
